### PR TITLE
Validate same filters in duplicate indexes

### LINKED ISSUE #24968: Validate that filters are the same for shared indexes
When using base class, attempting to be unique within the specific typed class does not apply the unique with Code-First.

Example excerpt:

``` C#
public class TestContext : DbContext
{
  DbSet<ImplementClassOne> Ones { get; set; }
  DbSet<ImplementClassTwo> Twos { get; set; }

  protected void OnModelCreating(ModelBuilder builder)
  {
    modelBuilder.Entity<Base>().HasDiscriminator<string>("Type")
      .HasValue<ImplementClassOne>("one")
      .HasValue<ImplementClassTwo>("two");

    modelBuilder.Entity<Base>().Property(base => base.UniqueVal)
      .IsRequired();

    modelBuilder.Entity<Base>().Property(base => base.RequiredVal)
      .IsRequired();

    modelBuilder.Entity<ImplementClassOne>().HasIndex(one => one.UniqueVal)
      .IsUnique()
      .HasFilter("Type = 'one'");

    modelBuilder.Entity<ImplementClassTwo>().HasIndex(two => two.UniqueVal)
      .IsUnique()
      .HasFilter("Type = 'two'");
  }
}
```

This configuration creates the following `Up()` method in the Migration file, using Code-First:
``` C#
protected override void Up(MigrationBuilder migrationBuilder)
{
  migrationBuilder.CreateTable(
    name: "Base",
    columns: table => new
    {
      Id = table.Column<int>(type: "int", nullable: false)
        .Annotation(...),
      UniqueVal = table.Column<string>(type: "nvarchar(max)", nullable: false,
      RequiredVal = table.Column<string>(type: "nvarchar(max)", nullable: false
    },
    constraints: table =>
    {
      table.PrimaryKey("PK_Base", x => x.Id);
    });

  migrationBuilder.CreateIndex(
    name: "IX_Base_UniqueVal_One",
    table: "Base",
    column: "UniqueVal",
    unique: true,
    filter: "Type = 'one'");
}
```

The second unique filtered index is missing. I can manually insert the index into the migration file. However, I am unable to test the second constraint when using normal unit testing methods, since unit tests normally do not use migrations.

Expected behavior would be for all unique filtered indexes be applied.

### PR DESCRIPTION
**Pull request by roji**
Fixes #24968

### CODE REVIEW

**Review by smitpatel — APPROVED** [review]

File: src/EFCore.Relational/Metadata/Internal/RelationalIndexExtensions.cs [hunk 1]
```diff
@@ -85,6 +85,20 @@ public static class RelationalIndexExtensions
                 return false;
             }
 
+            if (index.GetFilter() != duplicateIndex.GetFilter())
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Metadata/Internal/RelationalIndexExtensions.cs:88]:
```
            if (index.GetFilter(storeObject) != duplicateIndex.GetFilter(storeObject))
```

Comment (smitpatel) [reply src/EFCore.Relational/Metadata/Internal/RelationalIndexExtensions.cs:88]: /me goes and hides under a rock!

Comment (AndriySvyryd) [reply src/EFCore.Relational/Metadata/Internal/RelationalIndexExtensions.cs:88]: This allows to skip filter configuration for the index on the dependent in table sharing, avoiding a breaking change.

Comment (roji) [reply src/EFCore.Relational/Metadata/Internal/RelationalIndexExtensions.cs:88]: Oops, will submit a fixup.

Comment (roji) [reply src/EFCore.Relational/Metadata/Internal/RelationalIndexExtensions.cs:88]: #25939

**Pull request merged**